server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -968,7 +967,7 @@ public void writeTo(StreamOutput out) throws IOException {
         routingTable.writeTo(out);
         nodes.writeTo(out);
         if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
-            out.writeMap(transportVersions, StreamOutput::writeString, (o, v) -> TransportVersion.writeVersion(v, o));
+            out.writeMap(versionsWrappers, StreamOutput::writeString, (o, v) -> TransportVersion.writeVersion(v.transportVersion(), o));
```

Comment:
Likewise here could we make this `out.writeMap(versionsWrappers, StreamOutput::writeString, (o, v) -> v.writeTo(o);`